packages/augur-sdk/src/api/Gnosis.ts

```diff
@@ -305,32 +301,24 @@ export class Gnosis {
         setupData,
       });
 
-      this.safesToCheck.push({
-        safe: ethUtil.toChecksumAddress(response.safe),
-        owner: params.owner,
-        status: GnosisSafeState.WAITING_FOR_FUNDS,
-      });
-
-      this.augur.setGnosisStatus(GnosisSafeState.WAITING_FOR_FUNDS);
-
-      return response;
-    } catch(error) {
-      const restoreAddress = await this.calculateGnosisSafeAddress({
-        payment: '1',
+      const calculatedSafeAddress = await this.calculateGnosisSafeAddress({
+        ...response,
```

Comment:
It feels a bit dangerous to blindly use the response like this without validating that it contains reasonable data.  The `owner: params.owner` override I _think_ will protect against the obvious malicious relayer attack, but I'm mildly concerned that the relayer has the power to put basically whatever they want into this parameter.  "Probably fine", just a bit of a red flag.

Reply:
Digging a bit more into this, it appears that `calculateGnosisSafeAddress` has a `param` whose type is unnecessarily restrictive.  I think it should really be `params: { owner: string, payment: string}`.  With that change, you could then change the callsite to `calculateGnosisSafeAddress({ owner: params.owner, response.payment })`.  Though, it is unclear to me what `payment` is or why you need to get it from the response.

Reply:
@MicahZoltu The payment is amount we need to send to the safe address for the relay to create the safe. It is basically a "bid" on the part of the relay for their safe creation services.

Reply:
Ah, makes sense.  You may want to validate that the value you get back in there is a number (or whatever the expected type is) and potentially convert it to a `bigint` (or whatever you are using for big number storage in memory) rather tan leaving it as a string as you pass it around.